### PR TITLE
[W4.5c][F11-A2]Zhang Yijiong

### PR DESCRIPTION
**Pull request by Zhang-Y-J**
Inform the user if the storage file is set to Read-only

### CODE REVIEW

File: src/seedu/addressbook/storage/StorageFile.java [hunk 1]
```diff
@@ -106,12 +118,13 @@ public void save(AddressBook addressBook) throws StorageOperationException {
             marshaller.marshal(toSave, fileWriter);
 
         } catch (IOException ioe) {
-            throw new StorageOperationException("Error writing to file: " + path);
+            throw new StorageOperationException("Error writing to file: " + path + " file might be read-only");
```
Comment (karrui) [inline src/seedu/addressbook/storage/StorageFile.java:121]: Good job making the exception message more explicit to the user, but can `StorageOperationException` only be caused by a read-only error?

File: src/seedu/addressbook/storage/StorageFile.java [hunk 2]
```diff
@@ -52,6 +54,16 @@ public StorageOperationException(String message) {
         }
     }
 
+    /**
+     * Signals that some error has occured while trying to makes the storage file read only
+     * while the program is running
+     */
+    public static class StorageReadOnlyException extends Exception {
```
Comment (karrui) [inline src/seedu/addressbook/storage/StorageFile.java:61]: Extending from `Exception` may not be the best course of action, as all Exceptions may now be caught under `StorageReadOnlyException` instead. Maybe try to extend the already existing `StorageOperationException`?

**Review by karrui — COMMENTED** [review]
@ZhangYijiong Some comments added. Please close the PR after reading comments

**Comment by Zhang-Y-J** [comment]
Comments read.

**Pull request closed without merging**